static/app/utils/formatters.tsx

```diff
@@ -384,67 +384,90 @@ export function formatPercentage(value: number, places: number = 2) {
   );
 }
 
-const numberFormats = [
-  [1000000000, 'b'],
-  [1000000, 'm'],
-  [1000, 'k'],
-] as const;
-
 /**
- * Formats a number to a string with a suffix
+ * Formats a number with an abbreviation e.g. 1000 -> 1k.
  *
  * @param number the number to format
  * @param precision the number of significant digits to include
+ * @param includeDecimals when true, formatted number will always include non trailing zero decimal places
  */
 export function formatAbbreviatedNumber(
   number: number | string,
-  precision?: number
+  precision?: number,
+  includeDecimals?: boolean
 ): string {
   number = Number(number);
 
-  let lookup: (typeof numberFormats)[number];
+  const abbreviations = ['k', 'm', 'b'];
 
-  // eslint-disable-next-line no-cond-assign
-  for (let i = 0; (lookup = numberFormats[i]); i++) {
-    const [suffixNum, suffix] = lookup;
+  for (let i = 3; i >= 1; i--) {
+    const [suffixNum, suffix] = [Math.pow(10, i * 3), abbreviations[i - 1]] as const;
     const shortValue = Math.floor(number / suffixNum);
     const fitsBound = number % suffixNum;
```

Comment:
Hmm, if I understand it correctly the `fitsBound` is `0` (a falsy value) if it fits the bounds and any other number (truthy) if it doesn't. So  its boolean representation has exactly the opposite meaning.
I think we should change it to this:
```suggestion
    const fitsBound = number % suffixNum === 0;
```

Beware that in line 412, you need to remove the negation if you apply this:
```ts
    const useShortValue = !includeDecimals && (shortValue / 10 > 1 || fitsBound);
```